onnxruntime/core/providers/dnnl/dnnl_provider_factory.cc

```diff
@@ -32,6 +33,21 @@ std::unique_ptr<Provider_IExecutionProvider> DnnlProviderFactory::CreateProvider
 
 struct Dnnl_Provider : Provider {
   std::shared_ptr<Provider_IExecutionProviderFactory> CreateExecutionProviderFactory(int use_arena) override {
+#if defined(_WIN32) && !defined(_OPENMP)
```

Comment:
Do we need something for linux as well?

https://github.com/microsoft/onnxruntime/issues/4652

Reply:
Linux can use "-Wl,-z,nodelete" build flag to accomplish the same goal.